lib/modules/versioning/composer/index.spec.ts

```diff
@@ -12,8 +12,11 @@ describe('modules/versioning/composer/index', () => {
     a               | b                  | expected
     ${'1.2.0'}      | ${'v1.2'}          | ${true}
     ${'v1.0.0'}     | ${'1'}             | ${true}
+    ${'1.0alpha3'}  | ${'1.0.0-alpha.3'} | ${true}
```

Comment:
i don't think 1.0alpha3 is a valid composer version. does composer accept those?

Reply:
Yeah it seems to understand this format when testing, even if not recommended. Composer has a special `@stability` suffix that it uses for stability which it seemed to infer with 8.2rc1, but I'll think about ways I can test this more thoroughly.

Reply:
if it's not recommend, we should go the `php-version` datasource way and convert the versions there